lua/entities/gmod_tardis/modules/sh_falling.lua

```diff
@@ -0,0 +1,191 @@
+            if self:CallHook("ShouldNotPlayLandingSound") ~= true then
+                self:SendMessage((data.OurOldVelocity.z < -1500) and "fall_crashing_sound" or "fall_landing_sound")
+            end
+
+            self:SetData("vertbrakes", true)
+            self:Timer("vertbrakes", 1, function()
+                self:SetData("vertbrakes", false)
+            end)
+        end
+
+    end)
+
+	ENT:AddHook("PhysicsUpdate", "falling", function(self,ph)
+
+        local free_movement = not self:GetData("float") and not self:GetData("flight") and not self:IsPlayerHolding()
+        free_movement = free_movement and ph:IsGravityEnabled() and self:IsAlive()
+
+        self:SetData("free_movement", free_movement, true)
+
+        if not free_movement then
+            if self:GetData("vertbrakes") then
+                self:SetData("vertbrakes", false)
+            end
+            return
+        end
+
+        local phm=FrameTime()*66
+        local up=self:GetUp()
+        local ri2=self:GetRight()
+        local fwd2=self:GetForward()
+        local ang=self:GetAngles()
+        local vel=ph:GetVelocity()
+        local vell=ph:GetVelocity():Length()
+        local cen=ph:GetMassCenter()
+        local mass=ph:GetMass()
+        local lev=ph:GetInertia():Length()
+        local angv=ph:GetAngleVelocity()
+
+        local function align_in_flight()
+            local angmax = math.max(math.abs(ang.p), math.abs(ang.r))
+            local angvmax = math.max(math.abs(angv.x), math.abs(angv.y))
+
+            ph:SetAngleVelocityInstantaneous(Vector(angv.x * 0.95, angv.y * 0.95, angv.z))
+
+            if angvmax <= 50 and angmax > 10 then
+                ph:ApplyForceOffset( up * -ang.p * 10, cen - fwd2 * lev)
+                ph:ApplyForceOffset(-up * -ang.p * 10, cen + fwd2 * lev)
+                ph:ApplyForceOffset( up * -ang.r * 10, cen - ri2 * lev)
+                ph:ApplyForceOffset(-up * -ang.r * 10, cen + ri2 * lev)
+            end
+        end
+
+
+        local function reduce_movement()
+            if math.abs(ang.p) <= 35 and math.abs(ang.r) <= 35 and vel.z > 0 then
+                local newvel_x = math.Clamp(vel.x * 0.05,-30,30)
+                local newvel_y = math.Clamp(vel.y * 0.05,-30,30)
+
+                ph:SetVelocityInstantaneous(-up * 100 * phm)
+                ph:AddVelocity(Vector(newvel_x,newvel_y,0))
+
+                local newavel_x = math.Clamp(angv.x * 0.1,-300,300)
+                local newavel_y = math.Clamp(angv.y * 0.1,-300,300)
+                local newavel_z = math.Clamp(angv.z * 0.1,-300,300)
+                ph:SetAngleVelocityInstantaneous(Vector(newavel_x,newavel_y,newavel_z))
+            end
+        end
+
+        local pressing_down = IsValid(self.pilot) and TARDIS:IsBindDown(self.pilot,"flight-down")
+        local vertbrakes = self:GetData("vertbrakes")
+        local stopped = (vell < 0.1)
+
+        if pressing_down and not vertbrakes and vell > 5 then
```

Comment:
The velocity tolerance seems a bit too low here, you can have the TARDIS flat on it's face with the slightest amount of movement and holding CTRL will make it stand up. Maybe it should be downwards velocity and a bit higher or do we intend for sliding on a flat surface to be able to make it vertical?